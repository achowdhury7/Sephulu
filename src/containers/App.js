import React, { Component } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import { categoryActions, productActions, priceRangeActions } from '../actions'
import { Header } from '../components'
import { PageContent } from '../containers'


class App extends Component {
  componentDidMount() {
    const defaultPriceRanges = [
      { min: 0, max: 2500},
      { min: 2501, max: 5000},
      { min: 5001, max: 7500},
      { min: 7501, max: 10000},
    ]
    const defaultCategories = [ 'tools', 'brushes', 'markup']

    this.props.actions.setProductsFetching()
    this.props.actions.fetchProducts()
    this.props.actions.setPriceRanges(defaultPriceRanges)
    this.props.actions.setCategories(defaultCategories)
  }
  render() {
    return (
      <div>
        <Header />
        <PageContent />
      </div> 
    )
  }
}

const mapDispatchToProps = dispatch => 
  ({
    actions: bindActionCreators(Object.assign(
        {},
        categoryActions,
        productActions, 
        priceRangeActions),
      dispatch
    )
  })

const mapStateToProps = state => ({})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App)
