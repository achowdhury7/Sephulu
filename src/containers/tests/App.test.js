import React from 'react'
import { shallow, mount, render } from 'enzyme'

import { App } from '../App'

test('render app', () => {
  const mockActions = {
    setProductsFetching: jest.fn(),
    fetchProducts: jest.fn(),
    setPriceRanges: jest.fn(),
    setCategories: jest.fn()
  }
  const wrapper = shallow(
    <App actions={mockActions} />
  )
  expect(wrapper).toMatchSnapshot()
})