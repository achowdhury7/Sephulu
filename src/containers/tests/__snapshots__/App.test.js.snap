// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render app 1`] = `
<div>
  <Header />
  <Connect(PageContent) />
</div>
`;
